aws/cf.go

```diff
@@ -145,11 +142,75 @@ type healthCheck struct {
 	interval time.Duration
 }
 
+type Certificate struct {
+	CertificateArn string `yaml:"CertificateArn"`
+}
+
+// cfTemplate is an opauqe structure for unmarshaling a yaml cloudformation
+// stack into in order to replace the list of certificates for the
+// HTTPSListener.
+type cfTemplate struct {
+	AWSTemplateFormatVersion string      `yaml:"AWSTemplateFormatVersion"`
+	Description              string      `yaml:"Description"`
+	Parameters               interface{} `yaml:"Parameters"`
+	Conditions               interface{} `yaml:"Conditions"`
+	Resources                struct {
+		HTTPListener  interface{} `yaml:"HTTPListener"`
+		HTTPSListener struct {
+			Type       string `yaml:"Type"`
+			Condition  string `yaml:"Condition"`
+			Properties struct {
+				DefaultActions []struct {
+					Type           string `yaml:"Type"`
+					TargetGroupArn string `yaml:"TargetGroupArn"`
+				} `yaml:"DefaultActions"`
+				LoadBalancerArn string        `yaml:"LoadBalancerArn"`
+				Port            int           `yaml:"Port"`
+				Protocol        string        `yaml:"Protocol"`
+				Certificates    []Certificate `yaml:"Certificates"`
+			}
+		}
+		LB interface{} `yaml:"LB"`
+		TG interface{} `yaml:"TG"`
+	} `yaml:"Resources"`
+	Outputs interface{} `yaml:"Outputs"`
+}
+
+// injectCertificates injects a list of certificates into the cloudformation
+// template.
+func injectCertificates(cfTmpl string, certs []string) (string, error) {
```

Comment:
I am not happy with the solution parsing string, modify object and return string.
This looks more like a hack to me. We should have a type based object that can render to a cloudformation stack.